new-frontiers-14/frontier-station-14
# Add a signal receiver to thrusters (and gyros) for remote power control

## About the PR



This PR adds the ability to link Thrusters and Gyros to a button, lever or a switch. This is a followup to https://github.com/new-frontiers-14/frontier-station-14/pull/2025.

In order to have thruster respond to remote signal events, I implement the OnSignalReceived signal handler for it. I modify OnActivateThruster so that the message args are optional; this way I can call it from inside the thruster system without sending another event.

## Why / Balance


This is the problem I am trying to tackle as described in #2025. Basically, I want to be able to turn systems on and off regardless of the APC network it is attached to, so that I can save power. This way I can make a switch for the science room, one for thrusters, and so on. Based on https://github.com/new-frontiers-14/frontier-station-14/pull/2025#issuecomment-2356115048

If this is merged I will add this to a few more things, starting with entities that take a lot of power: arti analyzers, robotics console, etc.

## How to test


Spawn a shuttle and start linking things. Test to see if the engines actually stop working when toggled and check power savings.

## Media


![2024-09-17-135559_676x588_scrot](https://github.com/user-attachments/assets/b819e2d3-ebaa-47d1-9846-59ed934b8799)
![2024-09-17-135613_312x267_scrot](https://github.com/user-attachments/assets/c3a29218-0cf3-4123-bdb8-f8a982bad5d1)
![2024-09-17-135621_306x267_scrot](https://github.com/user-attachments/assets/c3f6fb23-fb2c-43e9-aed9-3232f9688b69)

## Requirements


## Breaking changes


Just adding an event handler and tweaking an existing event handler to be optionally called without event arguments.

**Changelog**

:cl:
- add: Thruster can now be switched on and off with remote buttons, switches, levers and signalers.
